src/pages/LoginModal/KakaoLogin.js

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { API_KEY, REDIRECT_URI } from './LoginModal';
+
+export const BASE_URL = `http://10.58.52.106:8000/users/login`;
+
+const KakaoLogin = () => {
+  const [searchParams] = useSearchParams();
+  const code = searchParams.get('code');
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    fetch('https://kauth.kakao.com/oauth/token', {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+      body: `grant_type=authorization_code&client_id=${API_KEY}&redirect_url=${REDIRECT_URI}&code=${code}`,
+    })
+      .then(res => res.json())
+      .then(res => {
+        if (res.access_token) {
+          fetch(`${BASE_URL}`, {
+            method: 'POST',
+            headers: {
+              authorization: res.access_token,
+            },
+          })
+            .then(res => res.json())
+            .then(data => {
+              if (data.accessToken) {
+                localStorage.setItem('token', data.accessToken);
+                navigate('/');
+                return;
+              }
+              alert('로그인 실패');
+            });
+        }
+      });
```

Comment:
불필요한 if문이 많아보입니다.
이렇게 바꿔보는건 어떨까요? ealry return 으로 방어코드를 만들어놓고 방어코드에 해당하는게 걸리면 빠르게 리턴해버리는 방식입니다.
```suggestion
      .then(({ access_token }) => {
        if (!access_token) return alert('다시 시도해주세요.')  // 방어코드
        
        fetch(`${BASE_URL}`, {
            method: 'POST',
            headers: { authorization: access_token },
          })
          .then(res => res.json())
          .then(({ accessToken }) => {
            if(!accessToken) return alert('로그인 실패');
            
            localStorage.setItem('token', data.accessToken);
            navigate('/');
          }
         });
        }
      });
```

Reply:
너무 많은 then chaining 이 일어나고 있어서 async await를 이용해도 좋을 것 같은 구문입니다. 
(필수 수정사항은 아니고 suggestion 입니다! 바꾸고 싶으시다면 저 멘션해서 코멘트 남겨주세요!! 옮기는 과정을 코멘트에 남겨놓겠습니다.)

사실 async await는 무서운 개념이 아니라 then chainging을 쉽게 풀어주는 symactic sugar입니다.

Reply:
@YunKukPark asynce await에 대해 알지 못해서 두렵지만,,,, 코멘트 남겨주시면 시도해보겠습니다. 감사합니다.